packages/react-search-ui-views/src/view-helpers/getUrlSanitizer.js

```diff
@@ -0,0 +1,33 @@
+const VALID_PROTOCOLS = ["http:", "https:"];
+
+/**
+ *
+ * @param {URL} URLParser URL interface provided by browser https://developer.mozilla.org/en-US/docs/Web/API/URL
+ * @param {String} currentLocation String representation of the browser's current location
+ */
+export default function getUrlSanitizer(URLParser, currentLocation) {
+  // This function is curried so that dependencies can be injected and don't need to be mocked in tests.
+  return url => {
+    if (!url) return url;
+
+    let parsedUrl;
+
+    try {
+      // Attempts to parse a URL as absolute
+      parsedUrl = new URLParser(url);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    try {
+      // Attempts to parse a URL as relative
+      parsedUrl = new URLParser(url, currentLocation);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    if (!parsedUrl) return "";
+
+    return VALID_PROTOCOLS.includes(parsedUrl.protocol)
+      ? parsedUrl.toString()
```

Comment:
How about returning the original `url` here instead, just to avoid the URL being changed? I could only find that it gets changed in the case of `http://example.com` becoming `http://example.com/` (adds trailing slash) which is not a big deal.

Reply:
That's a good point. Let me do that, it would also let me remove that additional check you mentioned in your other comment.

Reply:
https://github.com/elastic/search-ui/pull/481/commits/57655049d66b550aa135add7d8088571d37f4442